File: apps/api/AssetManagement.Api/DTOs/ImportDtos.cs
namespace AssetManagement.Api.DTOs;

public record ImportRowResult(
    int RowNumber,
    bool IsValid,
    List<string> Errors,
    Dictionary<string, string?> Data
);

public record ImportValidationResponse(
    string EntityType,
    int TotalRows,
    int ValidRows,
    int InvalidRows,
    List<ImportRowResult> Rows
);

public record ImportExecuteResponse(
    string EntityType,
    int Imported,
    int Skipped,
    int Failed,
    List<string> Errors
);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/ImportDtos.kt
package com.assetmanagement.api.dto

data class ImportRowResult(
    val rowNumber: Int,
    val isValid: Boolean,
    val errors: List<String>,
    val data: Map<String, String?>
)

data class ImportValidationResponse(
    val entityType: String,
    val totalRows: Int,
    val validRows: Int,
    val invalidRows: Int,
    val rows: List<ImportRowResult>
)

data class ImportExecuteResponse(
    val entityType: String,
    val imported: Int,
    val skipped: Int,
    val failed: Int,
    val errors: List<String>
)
